astroquery/wfau/core.py

```diff
@@ -0,0 +1,903 @@
+        if username is None or password is None or community is None:
+            pass
+        else:
+            self.login(username, password, community)
+
+    def _login(self, username, password, community):
+        """
+        Login to non-public data as a known user.
+
+        Parameters
+        ----------
+        username : str
+        password : str
+        community : str
+        """
+
+        # Construct cookie holder, URL opener, and retrieve login page
+        self.session = requests.session()
+
+        credentials = {'user': username, 'passwd': password,
+                       'community': ' ', 'community2': community}
+        response = self.session.post(self.LOGIN_URL, data=credentials)
+        if not response.ok:
+            self.session = None
+            response.raise_for_status()
+        if 'FAILED to log in' in response.text:
+            self.session = None
+            raise Exception("Unable to log in with your given credentials.\n"
+                            "Please try again.\n Note that you can continue "
+                            "to access public data without logging in.\n")
+
+    def logged_in(self):
+        """
+        Determine whether currently logged in.
+        """
+        if self.session is None:
+            return False
+        for cookie in self.session.cookies:
+            if cookie.is_expired():
+                return False
+        return True
+
+    def _args_to_payload(self, *args, **kwargs):
+        request_payload = {}
+
+        request_payload['database'] = kwargs.get('database', self.database)
+
+        programme_id = kwargs.get('programme_id', self.programme_id)
+
+        request_payload['programmeID'] = self._verify_programme_id(
+            programme_id, query_type=kwargs['query_type'])
+        sys = self._parse_system(kwargs.get('system'))
+        request_payload['sys'] = sys
+        if sys == 'J':
+            C = commons.parse_coordinates(args[0]).transform_to(coord.ICRS)
+            request_payload['ra'] = C.ra.degree
+            request_payload['dec'] = C.dec.degree
+        elif sys == 'G':
+            C = commons.parse_coordinates(args[0]).transform_to(coord.Galactic)
+            request_payload['ra'] = C.l.degree
+            request_payload['dec'] = C.b.degree
+        return request_payload
+
+    def _verify_programme_id(self, pid, query_type='catalog'):
+        """
+        Verify the programme ID is valid for the query being executed.
+
+        Parameters
+        ----------
+        pid : int or str
+            The programme ID, either an integer (i.e., the # that will get passed
+            to the URL) or a string using the three-letter acronym for the
+            programme or its long name
+
+        Returns
+        -------
+        pid : int
+            Returns the integer version of the programme ID
+
+        Raises
+        ------
+        ValueError if the pid is 'all' and the query type is a catalog.
```

Comment:
this would probably look messed up in sphinx, boldface for the full line, add a new line before ``if``.